src/json/json_interface.cpp

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************\
+
+Module: JSON Commandline Interface
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// JSON Commandline Interface
+
+#include "json_interface.h"
+
+#include <iostream>
+
+#include <util/cmdline.h>
+#include <util/exception_utils.h>
+#include <util/json.h>
+#include <util/message.h>
+
+#include "json_parser.h"
+
+/// Parse commandline options from \p json into \p cmdline
+static void get_json_options(const json_objectt &json, cmdlinet &cmdline)
+{
+  const jsont &arguments = json["arguments"];
+  if(!arguments.is_array())
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "array expected", "'arguments'");
+  }
+
+  for(const auto &argument : to_json_array(arguments))
+  {
+    if(argument.is_string())
+      cmdline.args.push_back(argument.value);
+    else
```

Comment:
I believe you could increase readability by doing
```
if(!argument.is_string())
{
  throw ...
}

cmdline.args.push_back(argument.value);
```
because it reduces the number of closing braces that immediately follow each other. This applies even more so down below, where you currently have three closing braces.